### ISSUE TITLE
Low scores

### ISSUE THREAD

**Issue by lucky-23** [salesforce/densecap#41]
Hello, when I did the evaluation using the pre-trained weights I got really low scores as compared to the ones reported in the paper.  Can you please specify why this is happening? I used Activity net dataset. Also I wanted to ask why sometimes models(not this model) generate same captions for all the testing set?(not related to this model). I am relatively new to the video captioning research so I would be thankful if you could kindly give me some information regarding this matter.

**Comment by LuoweiZhou** [comment]
@lucky-23 You may refer to this issue: https://github.com/salesforce/densecap/issues/16#issuecomment-519323542

**Comment by lucky-23** [comment]
Hi  @LuoweiZhou ,I ran the the evaluation code from ActivityNet 2017 using the pre-trained model that you released. I used the activity net dataset. However, the scores reported are still very low as given in the paper.Is the reported score based on the evaluation done on GT segments and not on the learned proposals? You said that the old evaluation code does not penalize the the proposals that do not overlap with GT segments but still the scores are still very low. Can you please explain why this is happening ? And how do I get the same score as reported in the paper?

**Comment by LuoweiZhou** [comment]
This codebase only includes training and evaluation on **learned proposals** as we stated in the README. In the paper, Tab. 1 corresponds to with learned proposals and Tab. 5 corresponds to with GT events.

**Comment by lucky-23** [comment]
@LuoweiZhou Thanks for replying back! I ran the evaluation code from ActivityNet 2017 using the pretrained weights that you released but still the scores which I got is very low as compared to those reported on the paper. Can you please tell me how to get the same scores as reported on the paper?

**Comment by LuoweiZhou** [comment]
@lucky-23 Could you elaborate in a little more detail how did you perform the evaluation? Also, what are the numbers you ended up getting. This could help me better troubleshooting.

**Comment by lucky-23** [comment]
Hi @LuoweiZhou I used the pretrained weights that you released and did the evaluation on the validation set using the Activitynet 2017 evaluation code as given in this link. https://github.com/ranjaykrishna/densevid_eval/tree/b8d90707984bf9c99454ba82b089006f14fb62b3. 
The score reported is:
tI0U : 0.3
CIDEr : 0.0014
Bleu_4 : 0.0014
Bleu_3 : 0.0040
Bleu_2 : 0.0094
Bleu_1 : 0.0242
Precision : 0.1076
ROUGE_L : 0.0231
METEOR : 0.0122
Recall: 0.01424
The scores I got is too low as compared to the ones reported in the paper.

**Comment by LuoweiZhou** [comment]
@lucky-23 The link to the 2017 evaluation code is actually this: https://github.com/LuoweiZhou/densevid_eval

**Comment by lucky-23** [comment]
HI @LuoweiZhou  I used your evaluation code as well but still the scores I got for activity net dataset is really low as compared to the paper. Is it because the code in written in python 2.7 and I have python 3 installed? Do you think this will cause any change to the result?

**Comment by LuoweiZhou** [comment]
I do not see why the python version would result in such a major difference. Here are the result outputs from my run and you might want to compare them with yours to locate there the bug possibly is: https://drive.google.com/drive/folders/1Sm9jixy6FocVG8Lkf3FQdcI7G0I8ZyLk?usp=sharing

**Comment by lucky-23** [comment]
Hi @LuoweiZhou I did the evaluation on your generated segments and I got the correct meteor score! There seems to be a bug in my code and I will look into it. And lastly can you please tell me how long did you train your model for( number of epochs)?  
Thanks alot😊

**Comment by LuoweiZhou** [comment]
@lucky-23 Glad to help. You can specify the number of epochs here: https://github.com/salesforce/densecap/blob/master/scripts/train.py#L77 Usually 20 epochs is good enough. I don't quite remember the exact training time though but it should just take hours.
